src/namespace.ts

```diff
@@ -39,8 +39,12 @@ export class Namespace {
     /**
      * Remove a socket from the namespace.
      */
-    removeSocket(wsId: string): Promise<boolean> {
-        return Promise.resolve(this.sockets.delete(wsId));
+    async removeSocket(wsId: string): Promise<boolean> {
+        this.channels.forEach(async (sockets, channel) => {
```

Comment:
forEach will not wait a promise resolving. It is like as without async/await

Reply:
This can be better implemented with [async js](https://caolan.github.io/async/v3/docs.html#each)?

Reply:
I believe no sense to use async/await in current implementation